src/steam/SteamHandler.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using Godot;
+
+/// <summary>
+///   Handles communicating between the general codebase and the Steam sdk requiring code.
+///   This ensures that non Steam versions of the game can be built with normal Godot tools.
+///   If Steam library is not loaded most methods in here do nothing to allow easily calling them without checking
+///   first.
+/// </summary>
+public class SteamHandler : Node, ISteamSignalReceiver
+{
+    private static SteamHandler instance;
+
+    private bool wePaused;
+
+    private ISteamClient steamClient;
+
+    public SteamHandler()
+    {
+        instance = this;
+    }
+
+    public static SteamHandler Instance => instance;
+
+    /// <summary>
+    ///   True if Steam has been properly loaded
+    /// </summary>
+    public bool IsLoaded => steamClient is { IsLoaded: true };
+
+    /// <summary>
+    ///   True if Steam loading was attempted and error messages can be retrieved
+    /// </summary>
+    public bool WasLoadAttempted { get; private set; }
+
+    public override void _Ready()
+    {
+        OnSteamInit();
+    }
+
+    public override void _Process(float delta)
+    {
+        steamClient?.Process(delta);
+        PauseMode = PauseModeEnum.Process;
```

Comment:
Why do you set the `PauseMode` for each frame?

Reply:
That's a mistake, I meant to put this in _Ready

Reply:
Fixed: https://github.com/Revolutionary-Games/Thrive/pull/2720/commits/9eae0ccf12d067f3ffdffe7f42eb47ed54aa9f8a